syntax = "proto3";

package example;

// Define the request and response message types
message HelloRequest {
  string name = 1;
}

message GoodbyeRequest {
  string name = 1;
}

message HelloResponse {
  string message = 1;
}

message GoodbyeResponse {
  string message = 1;
}

message TemperatureRequest {
    string location = 1;
}

message TemperatureResponse {
    float temperature = 1;
}

// Define the service
service Greeter {
  rpc SayHello (HelloRequest) returns (HelloResponse);
  rpc SayGoodbye (GoodbyeRequest) returns (GoodbyeResponse);
  rpc GetTemperature (TemperatureRequest) returns (TemperatureResponse);
}
